### ISSUE TITLE
Issue with view suffix on views Generated via bigquery_etl.glean_usage.GleanAppPingViews

### ISSUE THREAD

**Issue by whd** [mozilla/bigquery-etl#2499]
https://github.com/mozilla-services/mozilla-pipeline-schemas/commit/695ca208a5dc7a8abfdee0d6b9f9013d41db2560

This errored (but for possibly an unrelated reason):

```

                       -----Query Job SQL Follows-----                       

    |    .    |    .    |    .    |    .    |    .    |    .    |    .    |
   1:-- Generated by Generated via bigquery_etl.glean_usage.GleanAppPingViews
   2:CREATE OR REPLACE VIEW
   3:  `moz-fx-data-shared-prod.regrets_reporter.deletion_request`
   4:AS
   5:SELECT
   6:  *
   7:FROM
   8:  `moz-fx-data-shared-prod.regrets_reporter_ucs.deletion_request`
   9:
```

I think this is wrong and the computation of suffix for `regrets_reporter_ucs` is thinking `regrets_reporter` is its user-facing view, when in actuality `regrets_reporter` and `regrets_reporter_ucs` are two different user-facing namespaces.

**Comment by whd** [comment]
Side note: the query template has a redundant `Generated by` in it: `-- Generated by Generated via bigquery_etl.glean_usage.GleanAppPingViews`.

**Comment by wlach** [comment]
@whd that actually looks right to me-- `regrets_reporter` is the app-wide dataset, while `regrets_reporter_ucs` is the app-specific dataset. This is a little strange for an application with only one app id (like this one), but I believe this is the intended behaviour. Please reopen if you have concerns.

**Comment by whd** [comment]
Well this sounds like a problem: we have a collision between a non-glean app and a glean one that we didn't anticipate. Basically `regrets-reporter` already exists as a non-glean user-facing dataset so we shouldn't try to create one here, or need to figure something out. That's a separate issue I meant to address since bq deploys are currently blocked on this, but we can possibly fold the two together.

**Comment by whd** [comment]
To be explicit:
the app-wide dataset generated by https://github.com/mozilla-services/mozilla-pipeline-schemas/tree/generated-schemas/schemas/regrets-reporter-ucs conflicts with the already existing user-facing dataset generated by
https://github.com/mozilla-services/mozilla-pipeline-schemas/tree/generated-schemas/schemas/regrets-reporter

**Comment by whd** [comment]
I think we should probably special case to not generate the app-wide dataset for regrets-reporter-ucs, since there's only one app we don't need it and it conflicts with the existing dataset. I'm open to other solutions, however until this is resolved the BQ pipeline is stalled.

**Comment by wlach** [comment]
> I think we should probably special case to not generate the app-wide dataset for regrets-reporter-ucs, since there's only one app we don't need it and it conflicts with the existing dataset. I'm open to other solutions, however until this is resolved the BQ pipeline is stalled.

That sounds reasonable. Let me investigate what would be involved.

**Comment by jklukas** [comment]
> I think we should probably special case to not generate the app-wide dataset for regrets-reporter-ucs, since there's only one app we don't need it and it conflicts with the existing dataset. I'm open to other solutions, however until this is resolved the BQ pipeline is stalled.

I agree that sounds appropriate for the time being, assuming that the level of effort to implement that workaround is low.

Is the old legacy telemetry-based regrets reporter defunct? Do we no longer expect to receive data for it? If so, then I'd like to see us scope out what it would take to remove the `regrets-reporter` namespace and associated datasets; we could potentially archive existing data before removing the datasets.

**Comment by whd** [comment]
> Is the old legacy telemetry-based regrets reporter defunct? Do we no longer expect to receive data for it? If so, then I'd like to see us scope out what it would take to remove the `regrets-reporter` namespace and associated datasets; we could potentially archive existing data before removing the datasets.

I don't have the context to answer this question, but https://bugzilla.mozilla.org/show_bug.cgi?id=1740483 might be a good place to start asking these questions. https://docs.google.com/document/d/1a89FH0jjByAjKkl3BxNtLAgqDtZr2HTp9550t1ZyBww/edit#heading=h.otase6981ta outlines the process for deleting this data if we commit to that.

**Comment by jklukas** [comment]
Asked in https://bugzilla.mozilla.org/show_bug.cgi?id=1740483#c11

**Comment by wlach** [comment]
@whd We were already generating a `regrets_reporter` dataset in bigquery-etl see e.g. this copy of generated-schemas from 5 days ago:

https://github.com/mozilla/bigquery-etl/tree/a0f4c21132ea652d6ae5464dd04ea83e5b4da6ed/sql/moz-fx-data-shared-prod/regrets_reporter

That said, it looks like not generating the glean datasets is quite simple. I have a PR ready to go on that if we still want it.

**Comment by whd** [comment]
> @whd We were already generating a `regrets_reporter` dataset in bigquery-etl see e.g. this copy of generated-schemas from 5 days ago:

Yes, this was for the old regrets-reporter namespace. We shouldn't be trying to create it or populate it via glean machinery since it already existed as a non-glean user-facing namespace.
